ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -817,13 +817,19 @@ def runTest(self):
             self.log("Starting reachability state watch thread...")
             self.watching    = True
             self.light_probe = False
+            self.watch_dataplane = True # Flag for reachability_watcher.
+            self.watch_controlplane = True # Flag for reachability_watcher.
+            self.watcher_dp_is_stopped = threading.Event() # Waiter Event for the Watcher of dataplane state is stopped.
+            self.watcher_dp_is_running = threading.Event() # Waiter Event for the Watcher of dataplane state is running.
+            self.watcher_cp_is_stopped = threading.Event() # Waiter Event for the Watcher of controlplane state is stopped.
+            self.watcher_cp_is_running = threading.Event() # Waiter Event for the Watcher of control state is running.
```

Comment:
I feel you are over complicating the control required here with the thread event thing. An state enum (or even boolean protected by a lock would be sufficient).

Reply:
Boolean control does not ensure that reachability_watcher (or its parts: data and control watchers) fully stopped. Without Event controls, reachability_watcher would contulnue printing 
"Send %5d Received %5d ping DUT"
"Send %5d Received %5d t1->servers"
to the log till the end of its current iteration.

With Event controls, the tests waits explicitly till reachability_watcher stopped its current iteration.